### PR TITLE
Headers are encoded and decoded using latin1 charset #1450

### PR DESCRIPTION
**Pull request by LeOndaz**

### CODE REVIEW

**Review by carltongibson — COMMENTED** [review]
Refs #1450 — can you see the gunicorn PR linked there, for some tests we might add?

**Comment by LeOndaz** [comment]
I think there's no need to create a separate test for a mock request with headers with latin-1 characters, imo we could just add a latin-1 header value in the existing mocked requests. But I still doubt the usefulness of this considering that latin-1 is backwards compatible with ascii and everything should work the same. Opinions or corrections are welcome (this is my first contribution btw)

**Comment by carltongibson** [comment]
It's about stopping regressions in the future. If we add a test now we get notified when it breaks later.

**Comment by LeOndaz** [comment]
Thank you, I'm writing a test for latin1 now and I'll commit it

File: tests/test_http.py [hunk 1]
```diff
@@ -389,4 +402,4 @@ def test_middleware_caching(self):
             "django.core.handlers.base.BaseHandler.load_middleware"
         ) as super_function:
             AsgiHandler(scope)  # Second Handler
-            self.assertFalse(super_function.called)
+            self.assertFalse(super_function.called)
```
Comment (LeOndaz) [inline tests/test_http.py:405]: I think the IDE did this formatting change

**Comment by LeOndaz** [comment]
This took time, The travis-ci said that I'm not using the request object I've created and I wanted to fix that even tho using it is useless, I just wanted to pass some latin-1 headers to create a request and ensure it's working

**Review by carltongibson — APPROVED** [review]
Hi @LeOndaz — Thanks for this. Super.

**Pull request merged**